src/java/org/apache/cassandra/utils/JVMStabilityInspector.java

```diff
@@ -140,15 +140,26 @@ else if (t instanceof UnrecoverableIllegalStateException)
             if (t instanceof FSError || t instanceof CorruptSSTableException)
                 isUnstable = true;
 
-        fn.accept(t);
-
         // Check for file handle exhaustion
         if (t instanceof FileNotFoundException || t instanceof FileSystemException || t instanceof SocketException)
             if (t.getMessage() != null && t.getMessage().contains("Too many open files"))
                 isUnstable = true;
 
         if (isUnstable)
+        {
+            if (!StorageService.instance.isDaemonSetupCompleted())
+                FileUtils.handleStartupFSError(t);
             killer.killCurrentJVM(t);
```

Comment:
So if `isUnstable == true`, we're going to kill the JVM no matter what, but is the point of calling `handleStartupFSError()` that we log a better message when, in fact, we're not done with daemon setup? (It seems like both of the possible consumers call it as well when the setup isn't done, so this at least preserves the existing behavior through the reordering of this block w/ the `accept()` call...)

Reply:
I forget what actually caused me to do this... there was a reason I swear!

Reply:
I believe a test failed without this check

Reply:
I sadly do not remember where, but I feel like it would be python dtest as that is checks logs and makes sure they happen in the expected order.